pkg/diskmaker/diskmaker.go

```diff
@@ -180,8 +192,27 @@ func (d *DiskMaker) symLinkDisks(diskConfig *DiskConfig) {
 
 	for storageClass, deviceArray := range deviceMap {
 		for _, deviceNameLocation := range deviceArray {
+			// get PV creation lock which checks for existing symlinks to this device
+			pvLock, pvLocked, existingSymlinks, err := internal.GetPVCreationLock(
+				deviceNameLocation.diskNamePath,
+				d.symlinkLocation,
```

Comment:
So this will cause a lock to be acquired on all of the specified  devices and they will only be released when the whole function exits. This is usually a bad idea and can cause deadlocks, we should acquire lock on one device at a time and release the lock immediately after processing of that device is done.

Reply:
fixed